apps/prairielearn/src/sprocs/check_assessment_access.sql

```diff
@@ -26,28 +27,81 @@ DECLARE
+    end_date_from_override TIMESTAMP WITH TIME ZONE;
+    start_date_from_override TIMESTAMP WITH TIME ZONE;
+
 BEGIN
+    -- Check if the user has an entry in the assessment_access_policies table for this assessment_id.
+    -- If yes, get the end_date from the assessment_access_policies table, otherwise, use the end_date from the assessment_access_rules table.
+    SELECT aap.end_date , aap.credit , aap.start_date
+    INTO end_date_from_override , credit_from_override , start_date_from_override
+    FROM assessment_access_policies as aap
+    WHERE aap.assessment_id = check_assessment_access.assessment_id
+        AND (aap.user_id= check_assessment_access.user_id)
+        -- AND start_date <= check_assessment_access.date
+        -- AND end_date >= check_assessment_access.date
+    ORDER BY end_date DESC
+    LIMIT 1;
+
+    IF end_date_from_override IS NOT NULL THEN
+        authorized = TRUE;
+        credit = credit_from_override;
+        credit_date_string =  CASE
+            WHEN (credit_from_override > 0) THEN
+                credit_from_override::text || '%' || ' until ' || format_date_short(end_date_from_override, display_timezone)
+                ELSE '' END;
+        time_limit_min = (DATE_PART('epoch', end_date_from_override - now() - INTERVAL '31 seconds') / 60)::integer;
+        password = NULL;
+        mode = NULL;
+        seb_config = NULL;
+        show_closed_assessment = FALSE;
+        show_closed_assessment_score = TRUE;
+        active = TRUE;
+        active_access_rule_id = 0;
+    ELSE
     -- Choose the access rule which grants access ('authorized' is TRUE), if any, and has the highest 'credit'.
     SELECT
         caar.authorized,
         caar.exam_access_end,
-        aar.credit,
+        -- aar.end_date,
         CASE
-            WHEN aar.credit > 0 AND aar.active THEN
-                aar.credit::text || '%'
+            WHEN end_date_from_override IS NOT NULL THEN end_date_from_override
+            ELSE aar.end_date
+        END AS end_date,
+        -- aar.credit,
+        CASE
+            WHEN credit_from_override IS NULL THEN aar.credit
+            ELSE credit_from_override
+        END AS credit,
+        CASE
+            WHEN (aar.credit > 0 OR credit_from_override > 0) AND aar.active THEN
+                (CASE
+                    WHEN credit_from_override IS NULL THEN aar.credit::text || '%'
+                    ELSE credit_from_override::text || '%'
+                END)
                 || (CASE
-                        WHEN aar.end_date IS NOT NULL
-                        THEN ' until ' || format_date_short(aar.end_date, display_timezone)
-                        ELSE ''
-                    END)
+                        WHEN end_date_from_override IS NULL THEN ' until ' || format_date_short(aar.end_date, display_timezone)
+                        WHEN end_date_from_override IS NOT NULL THEN ' until ' || format_date_short(end_date_from_override, display_timezone)
```

Comment:
As far as I can tell, `end_date_from_override` is always `NULL` here because of the enclosing `IF`/`ELSE` condition?